CONTRIBUTING.md

```diff
@@ -57,13 +57,14 @@ file, and also have the [.NET 6.0](https://dotnet.microsoft.com/en-us/download/d
 installed.
 
 ```shell
-pip install -r requirements_dev.txt
+pip install .[dev,test]
+pip uninstall neo3-boa --yes
 dotnet tool install Neo.Express --version 3.5.20 -g
 dotnet tool install Neo.Test.Runner --version 3.5.17 -g
 ```
 
-> You can compile your smart contracts using your local Neo3-boa project by installing it with pip using the `--editable` 
-> flag, i.e. `pip install -e path/to/local/clone/of/Neo3-boa`.
+> Some tests will fail if you have the neo3-boa package installed. We only need the dependencies, that's why the 
```

Comment:
I know that there are 2 instances where this problem can occur:

1. Installing with `pip install .[dev,test]`
  1.1. run `python ./boa3_test/tests/run_unit_tests.py`
  1.2. you'll get a [`FileNotFoundError: [Errno 2] No such file or director` error](https://app.circleci.com/pipelines/github/CityOfZion/neo3-boa/2563/workflows/d176c8ea-4922-440e-a3f9-cd23c89b3858/jobs/6759)

2. Installing with `pip install -e .[dev,test]`
  2.1. run `python ./boa3_test/tests/run_unit_tests.py`
  2.2. some tests will work but others won't
  2.3. you'll get [a mix of different errors](https://app.circleci.com/pipelines/github/CityOfZion/neo3-boa/2566/workflows/dbb45274-82c4-4652-babb-75739e5fb7c3/jobs/6765/parallel-runs/0/steps/0-105) (Neo3-boa Internal errors, Cli not logging some messages, Files not being found).

Reply:
1. the caching part might need the cache to be invalidated once to solve it from expecting an old file (although it seems like your latest commit already doesn't trigger it anymore). The other error is clearly a failure to find a dependency from the `test` dependencies. I don't get that error locally when running `run_unit_test.py`.
2. my general view on this is; if it worked before switching to `pyproject.toml` then it should work after or something is missing/different in what `pyproject.toml` is producing. 
2.1 the first error 
    ```
    ======================================================================
    ERROR: boa3_test.test_cli (unittest.loader._FailedTest)
    ----------------------------------------------------------------------
    ImportError: Failed to import test module: boa3_test.test_cli
    Traceback (most recent call last):
      File "/usr/local/lib/python3.10/unittest/loader.py", line 470, in _find_test_path
        package = self._get_module_from_name(name)
      File "/usr/local/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
        __import__(name)
    ModuleNotFoundError: No module named 'boa3_test.test_cli'
    ```

   is a pyproject.toml error as we're not including that folder when packing. Therefore when trying to run this test in editable mode it can't find it. Adding `'boa3_test.test_cli'` to the `include` list solves that error.

   2.* if i modify `run_unit_test.py` to only run the tests that are failing as follows
     ```python
             failing_tests = [
              "test_generate_manifest_file_with_imported_event",
              "test_generate_nefdbgnfo_file_with_user_module_import",
              "test_generate_nefdbgnfo_file_with_user_module_name_import",
              "test_generate_with_user_module_import",
              "test_generate_with_user_module_import_with_project_root",
              "test_from_import_all",
              "test_from_import_user_module",
              "test_from_import_user_module_from_root_and_file_directories",
              "test_from_import_user_module_with_alias",
              "test_import_module_with_init",
              "test_import_user_class_inner_files",
              "test_import_user_module",
              "test_import_user_module_with_alias",
              "test_import_user_module_with_global_variables",
              "test_import_user_module_with_not_imported_symbols",
              "test_import_user_module_with_not_imported_variables",
              "test_import_variable",
              "test_variable_access_from_imported_module",
              "test_variable_from_imported_module",
              "test_metadata_importing_external_contract_before_metadata_method",
              "test_metadata_info_supported_standards_from_package",
              "test_metadata_info_supported_standards_with_imported_event",
              "test_global_variable_same_id_different_scopes",
              "test_cli_compile_fail_fast_false",
              "test_from_import_user_module_recursive_import",
              "test_import_user_module_recursive_import",
              "test_log_multiple_errors_on_import",
              "test_balance_of"
          ]
  
          for test in list_of_tests_gen(test_discover):
              if test._testMethodName in failing_tests:
                  suite.addTest(test)
     ```
     then the only test that fails is `test_balance_of`. Which suggests there is an issue with re-using globals, class variables or general setup that affect the other tests. Something along those lines and that is now exposed now that we install in editable mode.
     
     For the `test_balance_of` the error is 
    ```
         ======================================================================
    FAIL: test_balance_of (boa3_test.tests.compiler_tests.test_native.test_gas.TestGasClass)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/erik/code/neo3-boa/boa3_test/tests/compiler_tests/test_native/test_gas.py", line 103, in test_balance_of
        self.assertEqual(expected_results[x], invokes[x].result)
    AssertionError: 10 != 100000000
    ```
    which is caused by 
    https://github.com/CityOfZion/neo3-boa/blob/374916bbc087c8ca909c67ca20dfbcd6a58c64ae/boa3_test/test_drive/neoxp/command/utils/__init__.py#L20

    which will pass the quantity as `0,0000001` instead of `0.0000001` which then incorrectly gets parsed to `1` in `neo-debugger` here: <https://github.com/neo-project/neo-express/blob/e75c812d238f8727b844a82d0679e22d35df7319/src/neoxp/TransactionExecutor.cs#L386>
removing the line fixes the problem for this test (and all other tests still behave the same).